manager/controlapi/ca_rotation.go

```diff
@@ -104,171 +103,237 @@ func validateExternalCAURL(dialer *net.Dialer, tlsOpts *tls.Config, caURL string
 	if parsed.Scheme != "https" {
 		return errors.New("invalid HTTP scheme")
 	}
-	conn, err := tls.DialWithDialer(dialer, "tcp", getConnectionStringFromHost(parsed.Host), tlsOpts)
+	host, port, err := net.SplitHostPort(parsed.Host)
+	if err != nil {
+		return err
+	}
+	if port == "" {
```

Comment:
Oh hm... I didn't realize a port was required.  It also looks like it returns an error if there are too many colons, though, and that does not seem to be something `url.Parse` will return an error on:  https://play.golang.org/p/lAmzyII0-z.